benchmarks/mem/csrmv.bril

```diff
@@ -0,0 +1,268 @@
+    p: ptr<int> = ptradd data i;
+    store p v;
+    i: int = add i one;
+    jmp .loop_cond;
+.loop_exit:
+    free rng;
+    ret;
+}
+
+# sparse matrix dense vector multiplication
+# This DOES NOT allocate memory for the result!
+# Please allocate memory before calling it!
+@csr_spmv(rows: int, cols: int, csr_rowptr: ptr<int>, csr_colidx: ptr<int>, csr_values: ptr<int>, vec: ptr<int>, res: ptr<int>) {
+    one: int = const 1;
+    zero: int = const 0;
+    # initialize result vector
+    i: int = const 0;
+.loop_init_cond:
+    cond: bool = lt i rows;
+    br cond .loop_init_body .loop_init_exit;
+.loop_init_body:
+    p: ptr<int> = ptradd res i;
+    store p zero;
+    i: int = add i one;
+    jmp .loop_init_cond;
+.loop_init_exit:
+
+    # loop over all rows
+    rid: int = const 0;
+.loop_rows_cond:
+    cond: bool = lt rid rows;
+    br cond .loop_rows_body .loop_rows_exit;
+.loop_rows_body:
+    p: ptr<int> = ptradd csr_rowptr rid;
+    start: int = load p;
+    p: ptr<int> = ptradd p one;
+    end: int = load p;
+
+    # loop over non-zeros in a row
+    j: int = add start zero;
+    .loop_nnzs_cond:
+        cond: bool = lt j end;
+        br cond .loop_nnzs_body .loop_nnzs_exit;
+    .loop_nnzs_body:
+        p: ptr<int> = ptradd csr_colidx j;
+        cid: int = load p;
+        p: ptr<int> = ptradd csr_values j;
+        mat_val: int = load p;
+        p: ptr<int> = ptradd vec cid;
+        vec_val: int = load p;
+        p: ptr<int> = ptradd res rid;
+        acc: int = load p;
+        incr: int = mul mat_val vec_val;
+        acc: int = add acc incr;
+        store p acc;
+        j: int = add j one;
+        jmp .loop_nnzs_cond;
+    .loop_nnzs_exit:
+
+    rid: int = add rid one;
+    jmp .loop_rows_cond;
+.loop_rows_exit:
+    ret;
+}
+
+@print_arr(arr: ptr<int>, size: int) {
+    one: int = const 1;
+    i: int = const 0;
+.loop_cond:
+    cond: bool = lt i size;
+    br cond .loop_body .loop_exit;
+.loop_body:
+    p: ptr<int> = ptradd arr i;
+    v: int = load p;
+    print v;
+    i: int = add i one;
+    jmp .loop_cond;
+.loop_exit:
+    ret;
+}
+
+# main function
+# ARGS: 50 50 5
+@main(rows: int, cols: int, degree: int): int {
+    one: int = const 1;
+    rptr_len: int = add rows one;
+    nnz: int = mul rows degree;
+    csr_rowptr: ptr<int> = alloc rptr_len;
+    csr_colidx: ptr<int> = alloc nnz;
+    csr_values: ptr<int> = alloc nnz;
+
+    call @gen_uniform_csr rows cols degree csr_rowptr csr_colidx csr_values;
+    call @print_arr csr_rowptr rptr_len;
+    call @print_arr csr_colidx nnz;
+    call @print_arr csr_values nnz;
+
+    vec: ptr<int> = alloc cols;
+    call @gen_vec cols vec;
+    call @print_arr vec cols;
+
+    res: ptr<int> = alloc rows;
+    call @csr_spmv rows cols csr_rowptr csr_colidx csr_values vec res;
+    call @print_arr res rows;
+
+    free csr_rowptr;
+    free csr_colidx;
+    free csr_values;
+    free vec;
+    free res;
+
+    good: int = const 0;
+    ret good;
```

Comment:
Would you consider removing the return value of main? While I guess this hasn't been specified, afaik there isn't any implementation that will make use of this value. Most will silently ignore it like `brili` does.

`brilirs` rejects this program(I thought because of #91 but git blame predates that) which I think is fair?

Let me know what you think.

Reply:
I agree. Indeed there is no specifiction saying `main` shoud return a value to be used. 
Shall I make a new PR for this?

Reply:
Good idea! Yes please, a new PR would be great. (I suppose we should document this, and probably add a check in `brilck`…)

Reply:
Just created #293